docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,299 @@
+# ADR 0009 — Versioning and Release Branching for the OCM Monorepo
+
+* Status: Proposed
+* Deciders: Gergely Brautigam, Fabian Burth, Jakob Moeller, Gerald Morrison, Ilya Khandamirov, Piotr Janik, Matthias Bruns
+* Date: 2025-09-15
+
+## Context
+
+The repository contains multiple independently developed components, e.g., `ocm-cli`, `ocm-controller`, and a root/wrapper component `ocm` (final name to be decided. Acts as a root component referencing versions of the sub-components).
+
+We require a single, operational specification that defines how components are versioned, how release branches are created and maintained, and how CI publishes releases.
+
+Architectural requirement: deciding on a versioning strategy must include the implications for branching. Without an agreed branching model the versioning choice is incomplete and cannot be implemented safely.
+
+## Decision Drivers
+
+* Reproducible, auditable published releases.
+* Maintain per-component autonomy where practical.
+* Support straightforward hotfix and backport workflows for older minor releases.
+* Provide a machine-readable snapshot for documentation and website consumption.
+* Keep CI complexity manageable and deterministic.
+
+## Decision
+
+We adopt a combined, opinionated strategy:
+
+* Versioning: Hybrid approach — each component maintains a local `VERSION` file; the repository also maintains a generated `/VERSIONS.yaml` that mirrors the last published versions of all components (aka snapshot)
+* Branching: Component-scoped persistent release branches — branches follow the pattern `releases/<component>/v<major>.<minor>` and act as maintenance lines for each active minor release. Patches and hotfixes are applied via PRs to these branches.
+* Publishing: An annotated tag `<component>/v<major>.<minor>.<patch>` is the authoritative record of a published artifact.
+
+## SemVer Principles for Root Component
+
+### Normal Case (Automatic)
+
+The `ocm` root component version follows these automated rules for independent sub-component releases:
+
+* **Patch bump (x.y.z → z++)**: triggered by any sub-component patch release
+* **Minor bump (x.y.z → y++)**: triggered by any sub-component feature addition  
+* **Major bump (x.y.z → x++)**: triggered by any sub-component new major release
+
+Every OCM root release represents a reproducible snapshot of all sub-component versions and maintains strict SemVer compliance. The root component version is automatically computed and updated in `/VERSIONS.yaml` when any sub-component publishes a release.
```

Comment:
I think it could be better to always create RCs for the root component. that way, we never need to have the `supress-root-release` case you were defining in the edge case below. Instead everything would be a build candidate and we can choose any build candidate to be promoted.

Reply:
and then simply don't care about the new release that has been created in releases/ocm/v1.4 branch? By the way, we don't create PRs for Hotfixes, patches, do we? Then there would anyway be no RC for ocm for a submodule that triggered the patch. I'm confused :-)